pkg/middlewares/logger/gin/outgoing.go

```
@@ -0,0 +1,163 @@
+	logRequestBody  bool
+	logResponseBody bool
+	logHeaders      bool
+	fields          []logField
+}
+
+// NewOutgoingLogger creates a new outgoing request logger
+func NewOutgoingLogger(client *http.Client) *OutgoingLogger {
+	if client == nil {
+		client = http.DefaultClient
+	}
+	return &OutgoingLogger{
+		client: client,
+	}
+}
+
+// WithRequestBody enables request body logging
+func (o *OutgoingLogger) OutgoingWithRequestBody() *OutgoingLogger {
+	o.logRequestBody = true
+	return o
+}
+
+// WithResponseBody enables response body logging
+func (o *OutgoingLogger) OutgoingWithResponseBody() *OutgoingLogger {
+	o.logResponseBody = true
+	return o
+}
+
+// WithHeaders enables header logging
+func (o *OutgoingLogger) OutgoingWithHeaders() *OutgoingLogger {
+	o.logHeaders = true
+	return o
+}
+
+// WithField adds a custom field to the logger
+func (o *OutgoingLogger) WithField(key string, value interface{}) *OutgoingLogger {
+	o.fields = append(o.fields, logField{Key: key, Value: value})
+	return o
+}
+
+// WithFields adds multiple custom fields to the logger
+func (o *OutgoingLogger) WithFields(fields map[string]interface{}) *OutgoingLogger {
+	for k, v := range fields {
+		o.fields = append(o.fields, logField{Key: k, Value: v})
+	}
+	return o
+}
+
+// Do performs an HTTP request and logs the details using Gin context
+func (o *OutgoingLogger) Do(c *gin.Context, req *http.Request) (*http.Response, error) {
+	start := time.Now()
+
+	// Get trace ID from Gin context or request header
+	traceID := TraceIDFromContext(c)
+	if traceID == "" {
+		traceID = req.Header.Get("X-Trace-ID")
+	}
+
+	// Add trace ID to outgoing request headers
+	req.Header.Set("X-Trace-ID", traceID)
+
+	// Capture request body if enabled
+	var requestBody string
+	if o.logRequestBody && req.Body != nil {
+		bodyBytes, _ := io.ReadAll(req.Body)
+		requestBody = string(bodyBytes)
+		// Restore the body for the actual request
+		req.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
+	}
```

Ignoring errors for io.ReadAll